packages/typescript/internal/ts_project.bzl

```diff
@@ -389,6 +397,9 @@ def ts_project_macro(
         tsconfig = tsconfig,
         extends = extends,
         outdir = outdir,
+        # JSON files are special. They are output if outdir is set since tsc will
+        # copy them to outdir. Otherwise they are kind of both.
```

Comment:
good explanation 👍  if tsc copies them to outdir then they are outputs of the action, easy justification